.github/workflows/test_and_scan.yml

```diff
@@ -0,0 +1,82 @@
+# .github/workflows/ruby.yml
+
+name: Ruby Test and SonarCloud
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.0"
+          bundler-cache: true
+
+      - name: Lint via rubocop
+        run:
+          bundle exec rake -- rubocop
+
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ruby-version: ["3.0", "3.1", "3.2", "3.4"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+
+      - name: Test
+        run: |
+          bundle exec rake -- test
+
+      - name: Store coverage report
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-report
+          path: coverage/.resultset.json
+
+  sonar:
+    runs-on: ubuntu-20.04
+    needs: test   
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+    
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: "3.0" 
+        bundler-cache: true 
+
+    - name: Download coverage report
+      uses: actions/download-artifact@v3
+      with:
+        name: coverage-report
+        path: coverage
+
+    - name: Check
```

Comment:
Könnt später denke ich weg, oder?

Reply:
Yes, please :D Currently only for testing